### ISSUE TITLE
Python plugin for Memcachier

### ISSUE THREAD

**Issue by joseliber** [netdata/netdata#7147]
Hello everyone. I'm using the Memcached python plugin as a base to write a [Memcachier](https://www.memcachier.com/documentation/analytics-api) plugin.
To retrieve the data from Memcachier, I have to make two requests to this URL: https://analytics.memcachier.com/api/v1

First request will return a dict (actually, json parsed as dict) of `cache_id`.
With this ID, I can make the second request to:
https://analytics.memcachier.com/api/v1/{id}/stats

The result will have this structure:
```
{
  "a.b.c.totallyaserver.com:12345": {
    "limit_maxbytes": 0,
    "time": 0,
    "auth_cmds": 0,
    "auth_errors": 0,
    "bytes_read": 0,
    "bytes_written": 0,
    "cas_badval": 0,
    "cas_hits": 0,
    "cas_misses": 0,
    "cmd_delete": 0,
    "cmd_flush": 0,
    "cmd_get": 0,
    "cmd_set": 0,
    "cmd_touch": 0,
    "curr_connections": 0,
    "decr_hits": 0,
    "decr_misses": 0,
    "delete_hits": 0,
    "delete_misses": 0,
    "evictions": 0,
    "expired": 0,
    "get_hits": 0,
    "get_misses": 0,
    "incr_hits": 0,
    "incr_misses": 0,
    "total_connections": 0,
    "touch_hits": 0,
    "touch_misses": 0,
    "bytes": 0,
    "curr_items": 0,
    "total_items": 0
  }
}
```

Supposing I only have one cache_id and one _server_ (a.b.c.totallyaserver.com:12345 in the example above), I can just get the nested dict and send this to netdata as the response.

My question is, what would be the best way to make this work with multiple cache_id and/or multiple server.

The easiest way would be putting this information in the config file and have netdata retrive it from there. Then it will be `server*ids` jobs (each with one id, one server) for each run. The counterpoint for this is that the user would have to perform extra steps on the config file before having a functional plugin.

Please let me know your thoughts on this.

Regards,
José.

**Comment by ilyam8** [comment]
Hi. According [docs](https://www.memcachier.com/documentation/analytics-api)

> curl "https://analytics.memcachier.com/api/v1/login"
  --user CRED_USERNAME:CRED_PASSWORD

return 

> JSON response with memcachier ID

```json
{
    "cache_id": 1561
}
```

I guess it could contains several IDs, right?

**Comment by joseliber** [comment]
Hi @ilyam8, after further investigation, I've decided to limit the plugin to process only one cache_id and one cache_server for each job.
I did a test with Memcachier account that had many caches, and each cache has its own user:pass, so I'll expect to not deal with multiple servers/cache_ids.

I'm finishing the work on this plugin and hopefully I'll be able to submit a PR later this week.

I'm marking this as closed.

Regards,
José.
